packages/derived-wallet-solana/src/SolanaDerivedPublicKey.ts

```diff
@@ -54,25 +53,38 @@ export class SolanaDerivedPublicKey extends AccountPublicKey {
   }
 
   verifySignature({ message, signature }: VerifySignatureArgs): boolean {
-    const parsed = parseAptosSigningMessage(message);
-    if (!parsed || !(signature instanceof Ed25519Signature)) {
+    const parsedSigningMessage = parseAptosSigningMessage(message);
+    if (!parsedSigningMessage || !(signature instanceof Ed25519Signature)) {
       return false;
     }
 
-    const siwsInput = parsed.type === 'structuredMessage'
+    const signingMessageDigest = hashValues([message]);
+
+    // Obtain SIWS envelope input for the signing message
+    const siwsEnvelopeInput = parsedSigningMessage.type === 'structuredMessage'
       ? createSiwsEnvelopeForAptosStructuredMessage({
         solanaPublicKey: this.solanaPublicKey,
-        structuredMessage: parsed.structuredMessage,
+        structuredMessage: parsedSigningMessage.structuredMessage,
+        digest: signingMessageDigest,
       })
       : createSiwsEnvelopeForAptosTransaction({
         solanaPublicKey: this.solanaPublicKey,
-        aptosAddress: this._authKey.derivedAddress(),
-        rawTransaction: parsed.rawTransaction,
+        rawTransaction: parsedSigningMessage.rawTransaction,
+        digest: signingMessageDigest,
       });
 
-    const signInMessage = createSignInMessage(siwsInput);
+    // Matching the signature will ensure that the following fields are matching:
+    // - domain
+    // - solanaPublicKey
```

Comment:
this is also the account address
```suggestion
    // - solanaPublicKey (account address)
```